iroha_p2p/src/network.rs

```diff
@@ -165,10 +176,19 @@ where
     async fn handle(&mut self, msg: PeerMessage<T>) {
         match msg {
             PeerMessage::Connected(id) => {
-                let _ = self.connected_peers.insert(id);
+                #[allow(clippy::unwrap_used)]
+                let public_key = id.public_key.unwrap();
+                if self.connected_peers.contains(&public_key) {
+                    // TODO drop the peer
+                    warn!("Peer with public key {:?} already connected!", &public_key);
```

Comment:
Here we have to determine how to actually drop the actor that sent this message.